### PR TITLE
Update etcd server version to 3.2.14

### PR DESCRIPTION
**Pull request by jpbetz**
This upgrades the default etcd version used by kubernetes to 3.2.14

We previously [bumped the etcd client to 3.2.14](https://github.com/kubernetes/kubernetes/pull/57480).

Fixes https://github.com/kubernetes/kubernetes/issues/56438

```release-note
Upgrade default etcd server version to 3.2.14
```

cc @gyuho

### CODE REVIEW

**Comment by wojtek-t** [comment]
@jpbetz - can we do 3.2.14 instead - see: https://github.com/kubernetes/kubernetes/issues/58654

**Comment by jpbetz** [comment]
Thanks @ixdy. Don't know how I missed that! I'll grep for any other version occurrences sortly to make sure I didn't miss any others.

**Comment by jpbetz** [comment]
@wojtek-t Updating to 3.2.14 now to pick up the defrag fix.

**Comment by jpbetz** [comment]
In etcd logs on master: `etcdmain: error verifying flags, expected IP in URL for binding (https://e2e-17303-master:2380). See 'etcd --help'.`

**Comment by gyuho** [comment]
Is that `https://e2e-17303-master:2380` used as listen peer URL? Could be hitting this https://github.com/coreos/etcd/issues/6336, where we error in 3.3.

**Comment by jpbetz** [comment]
@gyuho Looks like it is, here's the exact flags passed to etcd:

```
/usr/local/bin/etcd --name etcd-e2e-17303-master 
--listen-peer-urls https://e2e-17303-master:2381 
--initial-advertise-peer-urls https://e2e-17303-master:2381 
--advertise-client-urls http://127.0.0.1:4002
--listen-client-urls http://127.0.0.1:4002 
--quota-backend-bytes=4294967296
--data-dir /var/etcd/data-events 
--initial-cluster-state new 
--initial-cluster etcd-e2e-17303-master=https://e2e-17303-master:2381 
--peer-trusted-ca-file /etc/srv/kubernetes/etcd-ca.crt
--peer-cert-file /etc/srv/kubernetes/etcd-peer.crt 
--peer-key-file /etc/srv/kubernetes/etcd-peer.key -peer-client-cert-auth
```

**Comment by wojtek-t** [comment]
It used to work with 3.1.*

**Comment by gyuho** [comment]
Can we change `s/--listen-peer-urls https://e2e-17303-master:2381/--listen-peer-urls https://127.0.0.1:2381/`?

[Reasoning behind this change](https://github.com/coreos/etcd/blob/master/Documentation/upgrades/upgrade_3_2.md#change-in---listen-peer-urls-and---listen-client-urls]):

> 3.2 now rejects domains names for --listen-peer-urls and --listen-client-urls (3.1 only prints out warnings), since domain name is invalid for network interface binding. Make sure that those URLs are properly formated as scheme://IP:port.

**Comment by jpbetz** [comment]
Binding to `127.0.0.1` in [cluster/gce/manifests/etcd.manifest](https://github.com/kubernetes/kubernetes/blob/58547ebd72bf314cba26e8d9148db282751e34f2/cluster/gce/manifests/etcd.manifest#L52) would be correct, right? The alternative would be to bind to `0.0.0.0` so that etcd binds to all interfaces, but we bind to `127.0.0.1` for `--listen-client-urls` for gce.

**Comment by jpbetz** [comment]
Oh, nevermind, just saw your comment @gyuho.  Yes, I'll try it out now.

**Comment by jpbetz** [comment]
`pull-kubernetes-e2e-gke-gci` is failing [across the board](https://prow.k8s.io/?repo=kubernetes%2Fkubernetes&job=pull-kubernetes-e2e-gke-gci), not just for this PR.

Closest matching issues I could find:

https://github.com/kubernetes/kubernetes/issues/58267
https://github.com/kubernetes/kubernetes/issues/58733

Neither appear to have been triaged, as of yet.

/retest

**Comment by jpbetz** [comment]
This is ready for review, PTAL.

**Comment by jpbetz** [comment]
cc @cheftako, @wenjiaswe

File: cluster/gce/manifests/etcd.manifest
Comment (cheftako) [inline cluster/gce/manifests/etcd.manifest]: Should advertise-client-urls really be to localhost?

File: staging/src/k8s.io/apiextensions-apiserver/artifacts/example/rc.yaml
Comment (cheftako) [inline staging/src/k8s.io/apiextensions-apiserver/artifacts/example/rc.yaml]: Should this be 3.2.14?

File: staging/src/k8s.io/sample-apiserver/artifacts/example/rc.yaml
Comment (cheftako) [inline staging/src/k8s.io/sample-apiserver/artifacts/example/rc.yaml]: Should this be 3.2.14?

File: cluster/gce/manifests/etcd.manifest
Comment (wojtek-t) [reply cluster/gce/manifests/etcd.manifest]: Yes - we don't want to expose it anywhere else.

Comment (wojtek-t) [reply cluster/gce/manifests/etcd.manifest]: Ohh, actually that's not fully true.
It's true for the client URL.

But I think that doing that for peer-urls will actually break e.g. GKE

Comment (wojtek-t) [reply cluster/gce/manifests/etcd.manifest]: @jpbetz - feel free to reach my offline to discuss this, but that will break things

File: staging/src/k8s.io/apiextensions-apiserver/artifacts/example/rc.yaml
Comment (jpbetz) [reply staging/src/k8s.io/apiextensions-apiserver/artifacts/example/rc.yaml]: Thanks for catching! This is what I get for starting with an upgrade to 3.2.13 and then switching to a upgrade to 3.2.14 after creating the initial PR.

File: cluster/gce/manifests/etcd.manifest
Comment (jpbetz) [reply cluster/gce/manifests/etcd.manifest]: @wojtek-t Yes, I think your (both) right. We should continue to advertise peer-urls via hostname. If we do change `--advertise-client-urls` to localhost, we would also need to update kube-apiserver's flags since it currently had ` --etcd-servers=http://127.0.0.1:2379`. I'll make that a separate change since it's not directly related to this PR and needs to be done carefully if we decide to do it.

File: staging/src/k8s.io/sample-apiserver/artifacts/example/rc.yaml
Comment (jpbetz) [reply staging/src/k8s.io/sample-apiserver/artifacts/example/rc.yaml]: Fixed, thanks for catching.

File: cluster/gce/manifests/etcd.manifest
Comment (wojtek-t) [reply cluster/gce/manifests/etcd.manifest]: So I think you need to revert the change this line, right?

**Comment by jpbetz** [comment]
rebasing

File: cluster/gce/manifests/etcd.manifest
Comment (jpbetz) [reply cluster/gce/manifests/etcd.manifest]: No, the bind to 127.0.0.1 is still correct. This discussion was only about advertise flags.

Comment (wojtek-t) [reply cluster/gce/manifests/etcd.manifest]: OK - I think you're right.

File: cluster/images/etcd/Makefile
Comment (cheftako) [inline cluster/images/etcd/Makefile]: Can we add a comment/warning about go version with etcd version? Is there a site which indicates which is the officially supported version of goland for each version of etcd. (A table of version https://github.com/coreos/etcd/releases/ which shows etcd 3.3 comes from go 1.9.2, **etcd 3.2 comes from 1.8.5**, ...)

**Review by cheftako — APPROVED** [review]
/lgtm but can we add the comment I indicated. Don't want that knowledge lost.

File: cluster/images/etcd/Makefile
Comment (jpbetz) [reply cluster/images/etcd/Makefile]: There is no tabular version, but I can add a comment to this makefile referencing the release page you linked to.

**Comment by jpbetz** [comment]
@cheftako Comment added. Good suggestion. This will need another lgtm since adding the comment changes the PR.

**Review by cheftako — APPROVED** [review]
/lgtm

**Review by cheftako — APPROVED** [review]
/lgtm

File: build/build-image/cross/Dockerfile
Comment (ixdy) [inline build/build-image/cross/Dockerfile]: we probably need to bump the crossbuild image for this to have an effect in the integration tests.

Comment (jpbetz) [reply build/build-image/cross/Dockerfile]: Thanks. I've updated `build/build-image/cross/VERSION` pushed `1.9.2-2`.

**Comment by jpbetz** [comment]
`pull-kubernetes-cross` build failure is due to https://github.com/kubernetes/kubernetes/pull/58174

**Comment by jpbetz** [comment]
I've applied #59126 manually to make sure all tests pass with the fix to cross. Looks like they did.

Once #59126 is merged I'll rebase and this should be ready to merge.

**Comment by jpbetz** [comment]
Rebasing off of https://github.com/kubernetes/kubernetes/pull/59126

**Comment by jpbetz** [comment]
@ixdy @wojtek-t @cheftako Looks like we've (finally) got this one ready to go.  Would you please take one last look? Thanks for all the feedback so far.

**Comment by lavalamp** [comment]
/approve

Under the understanding that this is a really low-impact update, and Joe is going to update documentation before 1.10 ships.

**Comment by jpbetz** [comment]
Thanks @lavalamp We'll use https://github.com/kubernetes/kubernetes/issues/59221 to track the doc changes.

**Comment by jpbetz** [comment]
Reapplying lgtm since only change was a rebase.

**Comment by jpbetz** [comment]
Rebasing to pick up godep changes.

**Comment by jpbetz** [comment]
`pull-kubernetes-cross` failing due to https://github.com/kubernetes/kubernetes/issues/59223 now. Will rebase again once the fix is merged.

**Comment by xiangpengzhao** [comment]
>We previously bumped the etcd client to 3.2.14.

@jpbetz I see that the linked PR https://github.com/kubernetes/kubernetes/pull/57480 just bumped the etcd client to 3.2.13, not 3.2.14. Am I missing something?

**Comment by jpbetz** [comment]
@xiangpengzhao When we created PR #57480, etcd 3.2.13 was the latest etcd 3.2.x version. We can bump the patch version as needed. It is fully compatible with the 3.2.14 server version, we don't require that the patch version of client and server match exactly.

**Comment by jpbetz** [comment]
Clean rebase. Reapplying lgtm.

**Comment by xiangpengzhao** [comment]
@jpbetz thanks for clarifying!

File: cluster/gce/manifests/etcd.manifest
Comment (anjensan) [reply cluster/gce/manifests/etcd.manifest]: It stops etcd nodes to communicate when they are running on different machines.
PR with revert - https://github.com/kubernetes/kubernetes/pull/59974

**Pull request merged**
